src/hooks/useFetchPlaylistData.ts

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState } from 'react'
+import { auth } from '@/api/firebaseApp'
+import { useFetchComments } from '@/hooks/useFetchComments'
+import { useFetchUserData } from '@/hooks/useFetchUser'
+import { useFetchBookmarks } from '@/hooks/useFetchBookmark'
+import { useToggleBookmark } from '@/hooks/useToggleBookmark'
+import { Playlist } from '@/components/playlist/Playlist'
+
+export const usePlaylistData = (playlist?: Playlist) => {
+  const [commentCount, setCommentCount] = useState(0)
+  const [isBookmarked, setIsBookmarked] = useState(false)
+  const [toastMessage, setToastMessage] = useState<string | null>(null)
+  const [isToastVisible, setIsToastVisible] = useState(false)
+  const [isOwner, setIsOwner] = useState(false)
+
+  const comments = useFetchComments(playlist?.id || '')
+  const userData = useFetchUserData(playlist?.userId)
+  const { mutate: toggleBookmark } = useToggleBookmark(playlist?.id || '')
+  const { data: BookmarkedData } = useFetchBookmarks(playlist?.id || '')
+
+  useEffect(() => {
+    if (comments) {
+      setCommentCount(comments.length)
+    }
+  }, [comments])
+
+  useEffect(() => {
+    if (BookmarkedData) {
+      setIsBookmarked(BookmarkedData.length > 0)
+    } else {
+      setIsBookmarked(false)
+    }
+  }, [BookmarkedData])
+
+  useEffect(() => {
+    const currentUser = auth.currentUser
+    if (currentUser && playlist) {
+      setIsOwner(currentUser.uid === playlist.userId)
+    }
+  }, [playlist])
+
+  const handleBookmark = () => {
+    if (isOwner) return
+
+    if (playlist && BookmarkedData !== undefined) {
+      toggleBookmark(isBookmarked)
+      setToastMessage(
+        !isBookmarked ? '북마크가 추가되었습니다.' : '북마크가 해제되었습니다.'
+      )
+      setIsToastVisible(true)
+    }
+  }
+
+  const hideToast = () => {
+    setIsToastVisible(false)
+  }
+
+  return {
+    commentCount,
+    userData,
+    isBookmarked,
+    handleBookmark: isOwner ? undefined : handleBookmark,
```

Comment:
P2;

owner인 경우 handleBookmark가 undefined로 나갈 것 같은데요. 이렇게 되면 hook을 사용하는 컴포넌트에서 undefined 대응을 해야 하게 됩니다.

handleBookmark 메서드에서 `if (isOwner) return` 이렇게 isOwner를 체크해주고 있기 때문에 굳이 undefined로 바꿔 return할 필요는 없지 않을까요?? (owner라면 handleBookmark 메서드는 empty return으로 종료되고 아무 일도 하지 않음)

```suggestion
    handleBookmark,
```

Reply:
안녕하세요! 이 부분을 삭제하게 되면, 페이지에서 내가 만든 플레이리스트는 아이콘이 안보이게 되는게 풀려버립니다.. ( 내 플레이리스트도 북마크가 보임 )
그래서 
 handleBookmark: isOwner ? undefined : handleBookmark,
  **->   handleBookmark,** 로 수정한 다음에

컴포넌트에 Playlist.tsx에 
  const {
    userData,
    isBookmarked,
    handleBookmark,
    toastMessage,
    isToastVisible,
    hideToast,
    **isOwner**
  } = usePlaylistData(playlist)

**->     isOwner 를 추가하고,**

**{!isOwner &&**
            (isBookmarked ? (
              <FaBookmark
                onClick={handleBookmark}
                css={filledbookmarkIconStyle}
              />
            ) : (
              <FaRegBookmark
                onClick={handleBookmark}
                css={bookmarkIconStyle}
              />
            ))}

**-> !isOwner 를 추가해서 감싸는 방식으로 변경했는데, 이렇게 바꿔도 되는지 여쭙니다!
화면상에서는 잘 되는 것 같아서 이렇게 했는데 혹시나 잘못된 방법일까봐 말씀드립니다!**

Reply:
네 어떤 의미인지 알았습니다.

기존:
handleBookmark가 있다면 북마크 아이콘도 그려주고, 아이콘 클릭시 handleBookmark가 실행도 됨

handleBookmark는 메서드이고, 우리가 일반적으로 메서드에 기대하는 것은 실행 가능성이지 "있음/없음" 등의 상태 및 값은 아니라고 생각합니다.

코드를 적게 짜는 것이 복잡도를 낮추는 것이 아닌데요.
의도가 명확히 드러나지 않는 방식으로 코드를 짜는 것은 복잡도를 높인다고 생각합니다.

메서드는 메서드 답게 실행만 하는게 맞다고 생각하고, owner 여부는 별도로 체크하는게 바람직해 보입니다.

Reply:
isOwner를 추가하는 방식으로 진행하는게 맞을 것 같아요.